ETL/ETL/_handle.h

```diff
@@ -385,16 +496,34 @@ class rhandle : public handle<T>
 		if(obj)add_to_rlist();
 	}
 
-	//! Default copy constructor
+	/** Copy constructor */
 	rhandle(const rhandle<value_type> &x):handle<T>(x)
 	{
 		if(obj)add_to_rlist();
```

Comment:
I wonder if the `rhandle::next_`, `rhandle::prev_`, `rshared_obj::front_` and `rshared_obj::back_` should be 'protected' by a mutex.

Reply:
I think yes. Since it is possible to access the same object from multiple threads, it is a good idea.